engine/src/main/java/org/camunda/bpm/engine/impl/db/sql/DbSqlSession.java

```diff
@@ -184,12 +184,19 @@ protected void bulkOperationPerformed(DbBulkOperation operation, int rowsAffecte
   }
 
   protected void entityDeletePerformed(DbEntityOperation operation, int rowsAffected, Exception failure) {
+    
     if (failure != null) {
       operation.setRowsAffected(0);
       operation.setFailure(failure);
 
+      DbOperation dependentOperation = operation.getDependentOperation();
```

Comment:
Let's try to get the wording right to avoid future confusion. I propose we call the parent operation (here: job update) the "dependency" and the child operation (here: byte array delete) the "dependent" (based on https://english.stackexchange.com/questions/25575/what-is-the-correct-word-for-dependee).